lib/mobility-core/src/Kernel/External/AadhaarVerification/Gridline/Types.hs

```diff
@@ -0,0 +1,115 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# LANGUAGE DerivingStrategies #-}
+
+module Kernel.External.AadhaarVerification.Gridline.Types where
+
+import Data.Aeson
+import EulerHS.Prelude hiding (error, state, (.=))
+
+data GridlineAadhaarOtpReq = GridlineAadhaarOtpReq
+  { aadhaar_number :: Text,
+    consent :: Text
```

Comment:
Enum can be created for this if you want as only allowed values are Y/N, fine otherwise too

Reply:
are you making it an enum ?

Reply:
i thing it is ok to pass it as Text because by default its value will be always Y only

Reply:
It's fine. He can make in next PR.